src/arkhamdb/ArkhamDb.ttslua

```diff
@@ -166,18 +166,20 @@ do
   ---@param playerColor String Color of the player this deck is being loaded for. Used for broadcast
   ---    if a weakness is added.
   internal.maybeDrawRandomWeakness = function(slots, playerColor)
-    local hasRandomWeakness = false
+    local randomWeaknessAmount = 0
     for cardId, cardCount in pairs(slots) do
       if cardId == RANDOM_WEAKNESS_ID then
         randomWeaknessAmount = cardCount
         break
       end
     end
-    if hasRandomWeakness then
-      local weaknessId = allCardsBagApi.getRandomWeaknessId()
-      slots[weaknessId] = 1
-      slots[RANDOM_WEAKNESS_ID] = nil
-      internal.maybePrint("Random basic weakness added to deck", playerColor)
+    if randomWeaknessAmount ~= 0 then
+      for i=1, randomWeaknessAmount do
+        local weaknessId = allCardsBagApi.getRandomWeaknessId()
+        slots[weaknessId] = 1
+        slots[RANDOM_WEAKNESS_ID] = nil
+      end
```

Comment:
Now that I read this again, I think this will result in too few weaknesses if the random weakness rolls the same one multiple times 😅
We're gonna need to check if the weaknessId was already added to the deck (`if slots[weaknessId] == 1`) and get a new one if it is already in there

Reply:
Hm and I'm not certain whether this will take the weakness count into consideration when importing multiple decks. Theoretically, it should since they are load one by one, but I am not sure if there is enough delay for the allcardsbag to pick up on the weaknesses in other peoples decks (and also doesn't work properly if a unique weakness is in a deck that will be loaded later)

Reply:
I'm probably going to target this in a separate PR since this one is already huge